public/pages/createDetector/components/DataFilters/SimpleFilter.tsx

```diff
@@ -76,6 +86,36 @@ export const SimpleFilter = (props: DataFilterProps) => {
   const lightModeStyles = {
     backgroundColor: '#F6F6F6',
   };
+
+  //If user search field name, filter filed names which include user's input word
+  //So user can only need to select from the filtered fileds.
+  const handleSearchFieldChange = debounce(async (searchValue: string) => {
+    let selectedFields: any = [];
+    if (searchValue) {
+      for (let i = 0; i < indexFields.length; i++) {
+        let selectedOptions: any = [];
+
+        let options = indexFields[i]?.options;
+        if (options) {
+          for (let j = 0; j < options?.length; j++) {
```

Comment:
Small: seems the `?` here is unnecessary since this is already in the `if (options)` code block

Reply:
Good catch, I refactored code before publish PR, forgot to remove this ?, will remove it.